scripts/deploy.js

```diff
@@ -4,26 +4,63 @@
 // When running the script with `hardhat run <script>` you'll find the Hardhat
 // Runtime Environment's members available in the global scope.
 const hre = require("hardhat");
+const delay = s => new Promise(res => setTimeout(res, s*1000));
 
 async function main() {
   await hre.run('compile');
 
   // We get the contract to deploy
   const dappStakingPoolFactory = await hre.ethers.getContractFactory("DappStakingPool");
-  const dappTokenFactory = await hre.ethers.getContractFactory("DappToken");
+  const funderFactory = await hre.ethers.getContractFactory("Funder");
 
-  const dappTokenContract = await dappTokenFactory.deploy();
-  await dappTokenContract.deployed();
-  console.log("Dapp Token deployed to:", dappTokenContract.address);
+  const dappStakingPoolProxy = await upgrades.deployProxy(dappStakingPoolFactory, [
+    "0x853c2D147a1BD7edA8FE0f58fb3C5294dB07220e", // liquidity protection
+    "0xf5FAB5DBD2f3bf675dE4cB76517d4767013cfB55", // liq protection store
+    "0x33A23d447De16a8Ff802c9Fcc917465Df01A3977", // dapp bnt anchor
+    "0x939b462ee3311f8926c047d2b576c389092b1649", // dapp token
+    "0x1f573d6fb3f13d689ff844b4ce37794d79a7ff1c", // bnt token
+    await hre.ethers.provider.getBlockNumber(), // start block
+    205000 // 20.5 DAPPs per block with 4 decimal precision
+  ]);
+  await dappStakingPoolFactory.attach(dappStakingPoolProxy.address);
```

Comment:
this line is not needed

Reply:
agreed https://github.com/ethers-io/ethers.js/blob/ce8f1e4015c0f27bf178238770b1325136e3351a/docs/v5/api/contract/example/README.md#erc20--attach-addressorname---contract

```
erc20 . attach( addressOrName ) => Contract
Returns a new instance of the Contract attached to a new address. This is useful if there are multiple similar or identical copies of a Contract on the network and you wish to interact with each of them.

```

So it would only be needed if the deployed contract was not the contract that was intended to be used.